client/backoff/backoff.go

```diff
@@ -68,6 +68,16 @@ func (b *Backoff) Retry(ctx context.Context, f func() error) error {
 	for {
 		err := f()
 		if err != nil {
+			// If the context has errored, return immediately.
+			// Can't just wait for Done() in this case, because
+			// this can race with the backoff, resulting in more
+			// calls to f() than expected.
+			if ctx.Err() != nil {
```

Comment:
If the context errors, does it invoke ctx.Done()? If it does, wouldn't the select return it first?